games/applications/run_bv.py

```diff
@@ -0,0 +1,65 @@
+# ------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the MIT License.
+# See accompanying file /LICENSE for details.
+# ------------------------------------------------------------------------------
+
+"""Benchmark Bernstein Vazirani circuits."""
+
+import pytest
+
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+from applications import backends, run_qiskit_circuit
+
+
+def build_bv_circuit(secret_string, mid_circuit_measure=False):
+    input_size = len(secret_string)
+    num_qubits = input_size + 1
+    if not mid_circuit_measure:
+        qr = QuantumRegister(num_qubits)
+        cr = ClassicalRegister(input_size)
+        qc = QuantumCircuit(qr, cr, name="main")
+        qc.x(qr[input_size])
+        qc.h(qr)
+        qc.barrier()
+        for i_qubit in range(input_size):
+            if secret_string[input_size - 1 - i_qubit] == "1":
+                qc.cx(qr[i_qubit], qr[input_size])
+        qc.barrier()
+        qc.h(qr)
+        qc.x(qr[input_size])
+        qc.barrier()
+        qc.measure(qr[:-1], cr)
+    else:
```

Comment:
This seems to me a bit more like a pre-optimization. Maybe we can look to push this into the compilers?

Reply:
I'd be cool with looking into getting the qiskit compiler to do this for us. Looking at the simulator output the fidelity is higher doing it with mid circuit measurement. I just based this on the two constructions that https://arxiv.org/abs/2110.03137 was doing for their BV benchmark